modeshape-jcr/src/main/java/org/modeshape/jcr/RepositoryConfiguration.java

```diff
@@ -1211,8 +1212,17 @@ public BinaryStore getBinaryStore() throws Exception {
                 String username = binaryStorage.getString(FieldName.USER_NAME);
                 String password = binaryStorage.getString(FieldName.USER_PASSWORD);
                 String bucketName = binaryStorage.getString(FieldName.BUCKET_NAME);
-                store = new S3BinaryStore(username, password, bucketName);
+
+                //Use S3 provided endpoints
+                if (FieldName.ENDPOINT_URL != null) {
```

Comment:
Hi @hchiorean I enabled allow null on the schema, so it branches depending of the use of this optional argument in the repo config and calls the appropriate overloaded constructor.  On my tests this segment is branching properly for both S3 (3 args) and S3 compatible storage (4 args).